### ISSUE TITLE
Calling `list().ofMinSize` with a large value causes `java.lang.IllegalArgumentException: bound must be positive` 

### ISSUE THREAD

**Issue by chriskiehl** [jlink/jqwik#377]
Howdy!

I'm seeing an unexpected `IllegalArgumentException` when passing larger values to `minSize`. 

Here's the failing test case:  

```
@Example
public void kaboom() {
    Arbitraries.strings()
            .list()
            .ofMinSize(1000)
            .sample();
}
```

This seems to happen for all arbitraries. Here's an example with `forType` and an object. 

```
@Value
static class ExampleObject {
    String a;
    String b;
}

@Example
public void kaboom() {
    Arbitraries.forType(ExampleObject.class)
            .useAllConstructors()
            .list()
            .ofMinSize(1000)
            .sample();
}
```

Both fail with the following error: 

```
  java.lang.IllegalArgumentException:
    bound must be positive

                              |-------------------jqwik-------------------
tries = 1                     | # of calls to property
checks = 1                    | # of not rejected calls
generation = RANDOMIZED       | parameters are randomly generated
after-failure = PREVIOUS_SEED | use the previous seed
when-fixed-seed = ALLOW       | fixing the random seed is allowed
edge-cases#mode = MIXIN       | edge cases are mixed in
edge-cases#total = 0          | # of all combined edge cases
edge-cases#tried = 0          | # of edge cases tried in current run
seed = 7233578493715904228    | random seed to reproduce generated values



java.lang.IllegalArgumentException: bound must be positive

	at java.base/java.util.Random.nextInt(Random.java:388)
	at net.jqwik.engine.properties.arbitraries.randomized.SizeGenerator.randomSize(SizeGenerator.java:59)
	at net.jqwik.engine.properties.arbitraries.randomized.SizeGenerator.lambda$sizeGeneratorWithCutoff$1(SizeGenerator.java:39)
	at net.jqwik.engine.properties.arbitraries.randomized.ContainerGenerator.next(ContainerGenerator.java:52)
	at net.jqwik.engine.facades.SampleStreamFacade.lambda$sampleStream$2(SampleStreamFacade.java:68)
	at net.jqwik.engine.facades.SampleStreamFacade.runInDescriptor(SampleStreamFacade.java:62)
	at net.jqwik.engine.facades.SampleStreamFacade.lambda$wrapInDescriptor$1(SampleStreamFacade.java:55)
	at java.base/java.util.stream.StreamSpliterators$InfiniteSupplyingSpliterator$OfRef.tryAdvance(StreamSpliterators.java:1360)
	at java.base/java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:127)
	at java.base/java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:502)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:488)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:474)
	at java.base/java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:150)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.base/java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:543)
	at net.jqwik.api.Arbitrary.sample(Arbitrary.java:424)
        ...
```

Jqwik version: 1.6.0

**Comment by jlink** [comment]
The exception I see in most recent version is:
```
  net.jqwik.api.JqwikException:
    minSize <1000> must not be larger than maxSize <255>.
```

This makes the problem and the workaround obvious:
```
Arbitraries.strings()
		   .list()
		   .ofMinSize(1000)
		   .ofMaxSize(1000)
		   .sample();
```

The open question: What would you expect to happen when you set minSize to a larger value than the default maxSIze?

**Comment by jlink** [comment]
> The open question: What would you expect to happen when you set minSize to a larger value than the default maxSIze?

Here's a few options I can think of:
1. `ofMinSize` will advance `maxSize` to the same number (only if `maxSize` was lower)
2. if (and only if) `maxSize` has not been explicitly specified `ofMinSize` will set `maxSize` to something like `minSize * 2`
3. `ofMinSize` will always set `maxSize` to something like `minSize * 2`, but `maxSize` can be set afterwards explicitly

Any other ideas?

**Comment by chriskiehl** [comment]
Aha, that error message clears up what was actually going on. Thanks! 

The behavior of option 2 or 3 sound the most intuitive to me as a newcomer to the library. My hunch going in was that specifying the `minSize` would automatically adjust the `maxSize` in some magical way (or that the upper bound be effectively limited only by the type itself (int)). But that's just a hot take from someone just taking their first steps with jqwik ^_^

**Comment by jlink** [comment]
If Integer,MAX were the default max size jqwik would mostly generate VERY large collections, which doesn’t help much in finding defects but takes a long time to shrink.

**Comment by jlink** [comment]
I'll go with option 2:

> if (and only if) `maxSize` has not been explicitly specified `ofMinSize` will set `maxSize` to something like `minSize * 2`

Otherwise, the same, more explicit error message should be given.

**Comment by jlink** [comment]
Fixed and released in 1.7.0-SNAPSHOT
